examples/deployment/haproxy/README.md

```diff
@@ -0,0 +1,107 @@
+# Deploying HAProxy Ingress Controller
+
+Don't have a Kubernetes cluster? Single-node of [CoreOS Kubernetes](https://github.com/coreos/coreos-kubernetes/) is a good starting point.
+
+Deploy a default backend used to serve `404 Not Found` pages:
+
+    kubectl run ingress-default-backend \
+      --image=gcr.io/google_containers/defaultbackend:1.0 \
+      --port=8080 \
+      --limits=cpu=10m,memory=20Mi \
+      --expose
+
+Check if the default backend is up and running:
+
+    kubectl get pod
+    NAME                                       READY     STATUS    RESTARTS   AGE
+    ingress-default-backend-1110790216-gqr61   1/1       Running   0          10s
+
+Deploy certificate and private key used to serve https on ingress that doesn't provide it's own certificate. For testing purposes a self signed certificate is ok:
+
+    openssl req \
+      -x509 -newkey rsa:2048 -nodes -days 365 \
+      -keyout tls.key -out tls.crt -subj '/CN=localhost'
+    kubectl create secret tls ingress-default-ssl --cert=tls.crt --key=tls.key
+    rm -v tls.crt tls.key
+
+Deploy HAProxy Ingress. Note that `hostNetwork: true` could be uncommented if your cluster has IPs that doesn't use ports 80, 443 and 1936.
```

Comment:
Suggest not making the default hostNetwork but instead just mentioning it as an alternative. hostNetwork might end up exposing more ports on the host than required, podnetwork + hostPort makes exposing each port a very conscious choice via pod yaml. If you want an example of hostport, see: https://github.com/kubernetes/ingress/blob/master/examples/deployment/nginx/nginx-ingress-controller.yaml#L38